docs/architecture/0004-eq-author-permissions-model.md

```diff
@@ -0,0 +1,64 @@
+# 4. EQ Author Permissions Model
+
+Date: 2019-03-11
+
+## Status
+
+Accepted
+
+## Context
+
+As part of the work to move Author into a live environment, in order to do that we have business requirements to fulfill around that to introduce controls around who can view and modify questionnaires in the system. Currently this is not locked down in any way, leading to undesired consequences, people can overwrite someone else's work and there's no clear audit trail at the moment about who made the change and what change was made.
+
+We need to solve these problems, and the simplest way we can begin to solve some of these problems is to introduce permissions based model to Author.
+
+## Decision
+
+We're proposing to implement a permissions model for Author with the following design.
+
+### Querying a questionnaire
+
+There are two parts to consider with this:
+
+1. Who has access to see the questionnaire.
+1. Who has access to make changes to a questionnaire.
+
+When the user logs in:
+
+1. Get the current user Id from the token.
+1. Loop through the questionnaire list.
+1. For each questionnaire check that the user has questionnaire read permission by looking it up in questionnaire list table.
+1. Show the questionnaire in the list if the user has read permissions, if the user doesn't have read permissions do not show questionnaire in the list and restrict them from being able to load the questionnaire.
+1. On loading a questionnaire if the user has write access allow them to make changes and save those changes to the questionnaire, if not only allow them to load the questionnaire but not make any changes.
+
+We plan to introduce read and write access for all initially and then encourage the users to remove/add read/write permissions to their questionniares for people who do or don't need to have it.
```

Comment:
Is this changing after the permission discussion? I thought it was now that owners will be only ones who can see things and then will add the necessary users.